### ISSUE TITLE
bug(stepper): Step error is not announced by screen readers

### ISSUE THREAD

**Issue by amysorto** [angular/components#23033]
The stepper component has an option to display errors in the header for a given step. The error appears when you navigate away from that step. This is only visual, and the error message is currently not read aloud by screen readers. The component should have a aria-live region to announce the error message politely.

**Comment by amysorto** [comment]
Announcing the step error in the component itself doesn't quite work since the stepper doesn't know the content is. With forms this can get a little tricky since the error might be related to a specific form field but the user won't be notified until they navigate to another step which can be jarring. Only developers have the context of the stepper content, so it should be handled by them instead. The docs have more details on best practices: https://material.angular.io/components/stepper/overview#forms.
